[await-timeout] Fix type definitions of set method

orangestark opened the pull request:
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).


If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: (https://github.com/vitalets/await-timeout/blob/12cd41e19cb5792ac2a994398f681e0deeec57c0/src/index.js#L29)
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

szhu commented on types/await-timeout/index.d.ts:
```diff
@@ -7,6 +7,6 @@ declare class Timeout {
 
     set(delay: number): Promise<undefined>;
-    set(delay: number, message: string): Promise<never>;
+    set(delay: number, message: string | (() => Error) | Error): Promise<never>;
```
Thanks, this looks correct!

However, it took me a while to follow the code in the library to be able to see that it's correct. Would it be possible for you to define an intermediate type for `toError` and reference that here? I think that would be helpful since it would make the code here more parallel to that of the library.

If you don't want to, that's okay, just let me know why. The code you have here is probably okay as is too.

orangestark replied:
Thank you for the quick and helpful feedback! I think it's great advice and I will make those adjustments accordingly.

DangerBotOSS commented:
<!--
  0 failure: 
  0 warning: 
  
  1 markdown notices
  DangerID: danger-id-Danger;
-->



## Formatting

The following files are not formatted:
1. types/await-timeout/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 603d4226d7efbd652ea320e1a0409d8064c32d43
</p>

orangestark commented:
Hi! I've added an intermediate type. Hope this is what you were looking for. I also fixed the argument type for static set, and added some cases in the test README. Let me know if there's problem. Thank you!

szhu reviewed (approved):
Perfect, thank you!

orangestark commented:
Ready to merge
